worker/rhpv2.go

```diff
@@ -329,83 +401,107 @@ func (s *Session) Read(ctx context.Context, w io.Writer, sections []rhpv2.RPCRea
 		MissedProofValues: newMissed,
 		Signature:         renterSig,
 	}
-	if err := s.transport.WriteRequest(rhpv2.RPCReadID, req); err != nil {
+	if err := s.transport.writeRequest(ctx, rhpv2.RPCReadID, req); err != nil {
 		return err
 	}
 
 	// host will now stream back responses; ensure we send RPCLoopReadStop
 	// before returning
-	defer s.transport.WriteResponse(&rhpv2.RPCReadStop)
+	defer s.transport.writeResponse(ctx, &rhpv2.RPCReadStop)
 	var hostSig *types.Signature
+
 	for _, sec := range sections {
-		// NOTE: normally, we would call ReadResponse here to read an AEAD RPC
-		// message, verify the tag and decrypt, and then pass the data to
-		// VerifyProof. As an optimization, we instead stream the message
-		// through a Merkle proof verifier before verifying the AEAD tag.
-		// Security therefore depends on the caller of Read discarding any data
-		// written to w in the event that verification fails.
-		msgReader, err := s.transport.RawResponse(4096 + uint64(sec.Length))
-		if err != nil {
-			return wrapResponseErr(err, "couldn't read sector data", "host rejected Read request")
-		}
-		// Read the signature, which may or may not be present.
-		lenbuf := make([]byte, 8)
-		if _, err := io.ReadFull(msgReader, lenbuf); err != nil {
-			return fmt.Errorf("couldn't read signature len: %w", err)
-		}
-		if n := binary.LittleEndian.Uint64(lenbuf); n > 0 {
-			hostSig = new(types.Signature)
-			if _, err := io.ReadFull(msgReader, hostSig[:]); err != nil {
-				return fmt.Errorf("couldn't read signature: %w", err)
+		errChan := make(chan error, 1)
```

Comment:
I don't think this needs to be buffered?

Also do we still need the goroutine with the timeout on the transport?

Reply:
> I don't think this needs to be buffered?

No, I usually make it a buffered channel if it's for something like this because I think it's more obvious what I'm doing. I know it'll only ever hold 0 or 1 errors, I'll make it non-buffered though if you dislike it. 

> Also do we still need the goroutine with the timeout on the transport?

`RawResponse` returns a custom reader object that's defined in the core package. Now that we're using deadlines instead of goroutines + contexts I guess we could do `setDeadline` and get rid of the goroutine